sdk/core/Azure.Core/src/HttpMessage.cs

```diff
@@ -113,6 +113,24 @@ internal void ApplyRequestContext(RequestContext? context, ResponseClassifier? c
             }
         }
 
+        internal void ApplyRequestContext(RequestContext? context, ResponseClassificationHandler? classificationHandler)
```

Comment:
On the code gen side, we would generate ResponseClassificationHandler rather than StatusCodeClassifier classes.